### PR TITLE
feature: reuse stale results

### PR DESCRIPTION
**Pull request by ghost**
This enables daml studio to goto definitions regardless whether the file
compiles or not.

### CODE REVIEW

**Comment by ghost** [comment]
Tested manually, works like a charm.

File: src/Development/IDE/Core/Rules.hs [hunk 1]
```diff
@@ -102,15 +103,17 @@ getDependencies file = fmap transitiveModuleDeps <$> use GetDependencies file
 getAtPoint :: NormalizedFilePath -> Position -> Action (Maybe (Maybe Range, [T.Text]))
 getAtPoint file pos = fmap join $ runMaybeT $ do
   opts <- lift getIdeOptions
-  spans <- useE GetSpanInfo file
-  return $ AtPoint.atPoint opts spans pos
+  (spans, mapping) <- useE GetSpanInfo file
+  !pos' <- MaybeT (return $ fromCurrentPosition mapping pos)
```
Comment (cocreature) [inline src/Development/IDE/Core/Rules.hs:107]: fwiw this is `hoistMaybe` in `errors`. Haven’t checked if we already depend on that (transitively). I’ll leave it up to you whether you want to use it (potentially adding the dependency) or not.

**Review by cocreature — APPROVED** [review]
Awesome, thanks! We should have a test for this in the daml repo. We have a similar one for stale codelenses iirc.

File: src/Development/IDE/Core/Rules.hs (hunk 1 above)
Comment (ghost) [reply src/Development/IDE/Core/Rules.hs:107]: I'll leave it like this to be consistent with upstream ghcide.

**Pull request merged**